.github/workflows/ecr-deletion-check.yaml

```diff
@@ -0,0 +1,56 @@
+on:
+  pull_request:
+    paths:
+      - 'namespaces/live.cloud-platform.service.justice.gov.uk/**'
+  workflow_dispatch:
+
+jobs:
+  ecr-deletion-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v44
+        with:
+          write_output_files: true
+
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+
+      - name: Get deletion protection from main
+        id: deletion-protection
+        env:
+          DELETED_FILES : ${{ steps.changed-files.outputs.deleted_files }}
+        run: |
+          DELETION_PROTECTION=
+          for file in $DELETED_FILES; do
+            if [[ "$file" =~ ^.*/resources/ecr.*\.tf$ ]]; then
```

Comment:
here you could have used `grep https://github.com/ministryofjustice/cloud-platform-terraform-ecr-credentials` instead of the if statement